src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/ScheduleService.kt

```diff
@@ -290,18 +313,32 @@ class ScheduleService(
   }
 
   private fun getSchedule15PcscMappings(): Triple<List<OffenceScheduleMapping>, List<OffenceScheduleMapping>, List<OffenceScheduleMapping>> {
-    val schedule15 = scheduleRepository.findOneByActAndCode("Criminal Justice Act 2003", "15")
-      ?: throw EntityNotFoundException("Schedule 15 not found")
-    val parts = schedulePartRepository.findByScheduleId(schedule15.id)
-    val part1Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 1 }.id)
+    val (part1Mappings, part2Mappings) = getSchedule15Mappings()
     val part1LifeMappings = part1Mappings.filter { it.offence.maxPeriodIsLife == true }
-    val part2Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 2 }.id)
     val part2LifeMappings = part2Mappings.filter { it.offence.maxPeriodIsLife == true }
     val seriousViolentOffenceMappings =
       part1Mappings.filter { it.paragraphNumber == "1" || it.paragraphNumber == "4" || it.paragraphNumber == "6" || it.paragraphNumber == "64" || it.paragraphNumber == "65" }
     return Triple(part1LifeMappings, part2LifeMappings, seriousViolentOffenceMappings)
   }
 
+  private fun getSchedule15Mappings(): Pair<List<OffenceScheduleMapping>, List<OffenceScheduleMapping>> {
+    val schedule15 = scheduleRepository.findOneByActAndCode("Criminal Justice Act 2003", "15")
+      ?: throw EntityNotFoundException("Schedule 15 not found")
+    val parts = schedulePartRepository.findByScheduleId(schedule15.id)
+    val part1Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 1 }.id)
+    val part2Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 2 }.id)
+    return Pair(part1Mappings, part2Mappings)
+  }
+
+  private fun getSexualOrViolentMappings(): Triple<List<OffenceScheduleMapping>, List<OffenceScheduleMapping>, List<OffenceScheduleMapping>> {
+    val scheduleThree = scheduleRepository.findOneByActAndCode("Sexual Offences Act 2003", "3")
+      ?: throw EntityNotFoundException("Schedule 3 not found")
+    val partThree = schedulePartRepository.findByScheduleId(scheduleThree.id)
```

Comment:
could replace lines 336 and 337 with a one line lookup 
`val scheduleThreeMapping = offenceScheduleMappingRepository.findBySchedulePartScheduleId(scheduleThree.id)`